crates/agent/src/controlplane.rs

```diff
@@ -242,13 +246,17 @@ impl<C: DiscoverConnectors + MakeConnectors> PGControlPlane<C> {
         let (ops_logs_template, ops_stats_template) =
             futures::try_join!(ops_logs_template, ops_stats_template)?;
 
+        let decrypted_hmac_keys = crate::decrypt_hmac_keys(&data_plane.encrypted_hmac_keys)
```

Comment:
We need these to be cached, we do a very high rate of these per second

Also, same comment applies here about preferring `hmac_keys` and using `encrypted_hmac_keys` only if they're not available. That gives us a very easy way to revert if we see problems during cutover.

Could we share the Snapshot approach, and have PGControlPlane keep a `Mutex<tables::DataPlanes>` ? If the requested `data_plane_id` is already in the table, then we're done. If it's not, then we would request _all_ data-planes, and decrypt hmac keys of (only) new ones.

Or even just ... put a Snapshot here? (perhaps with some flags to skip loads of collections & tasks, as they're chunky and not needed)